src/sardana/macroserver/macros/standard.py

```diff
@@ -845,3 +845,27 @@ def run(self, nr):
                 self.__loop()
                 progress = ((i + 1) / float(nr)) * 100
                 yield progress
+
+
+class newfile(Macro):
+    """This macro sets a new ScanFile and ScanID values into the env.
+    The ScanID should be set to the value N of the last executed ScanID
+    in order to continue with a ScanID N+1 for the next scan."""
+
+    env = ('ScanFile', 'ScanID')
+
+    param_def = [
```

Comment:
I would try to follow the spec command parameters order i.e. `ScanFile_list` first and then `ScanID`. In order to make the macros execution as user friendy as possible this would require extending the _spock syntax_. More details in #781.

Reply:
I agree on that.